idd/Energy+.idd.in

```diff
@@ -792,18 +792,22 @@ ZoneAirMassFlowConservation,
        \memo infiltration object mass flow rates.
        \memo If either mixing or infiltration is active, then the zone air mass flow
        \memo balance calculation will attempt to enforce conservation of mass for each zone.
-       \memo If mixing is "No" and infiltration is "None", then the zone air mass flow
+       \memo If zone flow balance method is "None" and infiltration is "None", then the zone air mass flow
        \memo calculation defaults to assume self-balanced simple flow mixing and infiltration objects.
        \unique-object
        \min-fields 3
-  A1,  \field Adjust Zone Mixing For Zone Air Mass Flow Balance
-       \note If Yes, Zone mixing object flow rates are adjusted to balance the zone air mass flow
-       \note and additional infiltration air flow may be added if required in order to balance the
-       \note zone air mass flow.
-       \type choice
-       \key Yes
-       \key No
-       \default No
+  A1,  \field Zone Air Mass Flow Balance Method
+       \note If AdjustZoneMixingFlow, Zone mixing object flow rates are adjusted to balance the zone air 
+       \note mass flow and additional infiltration air flow may be added if required in order to balance 
+       \note the zone air mass flow.
+	   \note If AdjustZoneReturnFlow, zone return air mass flow is adjusted to balance the zone air flow 
+	   \note and additional infiltration air flow may be added if required in order to balance the zone 
+       \note air mass flow. ZoneMixing objects air flow is always maintained at user specified value.	   
+       \type choice
+       \key AdjustZoneMixingFlow
+	   \key AdjustZoneReturnFlow
+       \key None
```

Comment:
It's a little confusing here how this field relates to the next one for Infiltration Balancing Method. If I'm following correctly "None"  in this field could still do infiltration balancing if the next field is not None. Maybe a better key choice or a more specific field name to indicate this field only applies to the mixing or return flow adjustment?

Reply:
Infiltration treatment applies to both zone air flow balancing methods. Currently If zone air flow balancing method is "None", then it won't apply infiltration treatment method for zone air flow balancing.

Do you want to apply infiltration treatment method regardless of the zone balancing method if infiltration treatment method not "None"? I can it change it easily?

Reply:
Yes, that is a current capability, and it needs to continue to be an option. 
That's why I think the modified field should be named differently (Adjust Zone Mixing or Return, Mixing, Return, Neither?) or broken into two fields, one for mixing and one for return if it's possible to have both active (say with one having priority over the other)?

Reply:
OK, I will maintain infiltration treatment method as an option.

Reply:
I think for now let's consider the two methods as separate input field but only one method can be active in a simulation. See below a modified ZoneAirMassFlowConservation object:

```
ZoneAirMassFlowConservation,
       \memo Enforces the zone air mass flow balance by adjusting zone mixing object and/or
       \memo infiltration object mass flow rates, or alternatively by adjusting zone return and/or
       \memo infiltration object mass flow rates. If both adjust the zone mixing and return flow input 
       \memo fields are set to "Yes", then Adjust the "Adjust Zone Return" method has the priority.  
       \memo If either zone mixing, zone return, or infiltration is active, then the zone air mass flow
       \memo balance calculation will attempt to enforce conservation of mass for each zone.
       \memo If adjust zone mixing is "No", adjust zone return is "No" and infiltration is "None", then 
       \memo zone air mass flow calculation defaults to assume self-balanced simple flow mixing and 
       \memo infiltration objects.
       \unique-object
       \min-fields 3  
  A1,  \field Adjust Zone Mixing For Zone Air Mass Flow Balance
       \note If Yes, Zone mixing object flow rates are adjusted to balance the zone air mass flow
       \note and additional infiltration air flow may be added if required in order to balance the
       \note zone air mass flow.
       \type choice
       \key Yes
       \key No
       \default No   
  A2,  \field Adjust Zone Return For Zone Air Mass Flow Balance
       \note If Yes, Zone return node flow rates are adjusted to balance the zone air mass flow
       \note and additional infiltration air flow may be added if required in order to balance the
       \note zone air mass flow.	   
       \type choice
       \key Yes
       \key No
       \default No     
  A3,  \field Infiltration Balancing Method
       \note This input field allows user to choose how zone infiltration flow is treated during
       \note the zone air mass flow balance calculation.
       \type choice
       \key AddInfiltrationFlow
       \key AdjustInfiltrationFlow
       \key None
       \default AddInfiltrationFlow



```

Reply:
@mjwitte Do you have preference which zone air mass balance method will have priority if user specifies "Yes" to both methods?

Reply:
Hmm, interesting question. My guess is that different applications may require one over the other. Would it be too messy to allow both?
Adjust Zone Mixing
\key No
\key Yes
\key YesAfterReturn
\key Secondary
Adjust Return
\key No
\key Yes
\key YesAfterMixing

Or go back to a single field "Adjust Zone Mixing and Return For Air Mass Flow Balance"?
\key AdjustMixingOnly
AdjustReturnOnly
AdjustMixingThenReturn
AdjustReturnThenMixing
None

Reply:
Allowing both is going to be too messy. By the way I have already made code changes to use the two field option but only one method (AdjustMixingOnly or AdjustReturnOnly) is allowed at a time. 

I can try the single field option with different combination if you have strong preference over the two field option. @mjwitte ?

Reply:
OK. I am going with this option because it will give flexibility when one method is not sufficient to balance the zone.

Or go back to a single field "Adjust Zone Mixing and Return For Air Mass Flow Balance"?
```
\key AdjustMixingOnly
\key AdjustReturnOnly
\key AdjustMixingThenReturn
\key AdjustReturnThenMixing
\key None
```

Reply:
Below is the modified ZoneAirMassFlowConservation object:
```
ZoneAirMassFlowConservation,
       \memo Enforces the zone air mass flow balance by either adjusting zone mixing object flow only, 
       \memo adjusting zone return flow only, adjusting zone mixing and then the zone return flows, 
       \memo or adjusting zone return and then the zone mixing object flows. Zone infiltration flow air 
       \memo is adjusted depending user selection in the infiltration treatment method input field  
       \memo If either of the zone mixing or return flow adjusting methods or infiltration is active, 
       \memo then the zone air mass flow balance calculation will attempt to enforce conservation of 
       \memo mass for each zone. If flow balancing method is "None" and infiltration is "None", then the
       \memo zone air mass flow calculation defaults to assume self-balanced simple flow mixing and 
       \memo infiltration objects.
       \unique-object
       \min-fields 3
  A1,  \field Adjust Zone Mixing and Return For Air Mass Flow Balance
       \note If "AdjustMixingOnly", zone mixing object flow rates are adjusted to balance the zone air 
       \note mass flow and additional infiltration air flow may be added if required in order to balance
       \note the zone air mass flow. If "AdjustReturnOnly", zone return node flow rates are adjusted to 
       \note balance the zone air mass flow and additional infiltration air flow may be added if required 
       \note in order to balance the zone air mass flow. If "AdjustMixingThenReturn", first Zone mixing 
       \note object flow rates are adjusted to balance the zone air and then the zone return node flow 
       \note rates are adjusted to balance the zone air mass flow and additional infiltration air flow 
       \note may be added if required in order to balance the zone air mass flow.
       \note If "AdjustReturnThenMixing", first zone return node flow rates are adjusted to balance the 
       \note zone air and then zone mixing object flow rates are adjusted to balance the zone air mass
       \note flow and additional infiltration air flow may be added if required in order to balance
       \note the zone air mass flow.	   
       \type choice
       \key AdjustMixingOnly
       \key AdjustReturnOnly
       \key AdjustMixingThenReturn
       \key AdjustReturnThenMixing
       \key None
       \default None
```